File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Assorted/CloseUnlocksFileTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4objects.Db4o;
using Db4objects.Db4o.Foundation.IO;
using Db4objects.Db4o.Tests.Common.Api;

namespace Db4objects.Db4o.Tests.Common.Assorted
{
	public class CloseUnlocksFileTestCase : Db4oTestWithTempFile
	{
		[System.ObsoleteAttribute]
		public virtual void Test()
		{
			File4.Delete(TempFile());
			Assert.IsFalse(Exists(TempFile()));
			IObjectContainer oc = Db4oEmbedded.OpenFile(NewConfiguration(), TempFile());
			oc.Close();
			File4.Delete(TempFile());
			Assert.IsFalse(Exists(TempFile()));
		}

		private bool Exists(string fileName)
		{
			return new Sharpen.IO.File(fileName).Exists();
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/assorted/CloseUnlocksFileTestCase.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.db4ounit.common.assorted;

import java.io.*;

import com.db4o.*;
import com.db4o.db4ounit.common.api.*;
import com.db4o.foundation.io.*;

import db4ounit.*;


public class CloseUnlocksFileTestCase extends Db4oTestWithTempFile {
    
	/**
	 * @deprecated
	 */
    public void test(){
        File4.delete(tempFile());
        Assert.isFalse(exists(tempFile()));
        ObjectContainer oc = Db4oEmbedded.openFile(newConfiguration(), tempFile());
        oc.close();
        File4.delete(tempFile());
        Assert.isFalse(exists(tempFile()));
    }

	private boolean exists(final String fileName) {
		return new File(fileName).exists();
	}

}
